rsocket-tck-drivers/src/main/java/io/rsocket/tckdrivers/test/TckTest.java

```diff
@@ -0,0 +1,149 @@
+package io.rsocket.tckdrivers.test;
+
+import io.rsocket.RSocket;
+import io.rsocket.RSocketFactory;
+import io.rsocket.tckdrivers.client.JavaClientDriver;
+import io.rsocket.tckdrivers.common.ServerThread;
+import io.rsocket.transport.netty.client.TcpClientTransport;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.net.URI;
+import java.util.*;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class TckTest {
+
+  /*
+   * Start port. For every input test file a server instance will be launched.
+   * For every server instance currentPort is incremented by 1
+   */
+  private static int currentPort = 4567;
+
+  private static final String hostname = "localhost";
+
+  /* Run all the test from "path". Tests files are expected to have same names
+   * with a prefix "server" or "client" to indicate whether they type.
+   */
+  private static final String path = "src/test/resources/";
+
+  private static final String serverPrefix = "server";
+  private static final String clientPrefix = "client";
+
+  private String name; // Test name
+  private JavaClientDriver jd; // javaclientdriver object for running the given test
+  private List<String> test; // test instructions/commands
+
+  public TckTest(String name, JavaClientDriver jd, List<String> test) {
+    this.name = name;
+    this.jd = jd;
+    this.test = test;
+  }
+
+  /** Runs the test. */
+  @Test
```

Comment:
add a timeout here, when I made a test fail it caused the whole test suite to hang.

![image](https://user-images.githubusercontent.com/231923/28059495-59134826-661c-11e7-8153-fef91721b366.png)